### PR TITLE
More Pubsub cleanup

### PR DESCRIPTION
**Pull request by sduskis**
- Removing PublisherStats, since it's package private and not used anywhere.
- StatusUtil and SubscriberStats are package private.  Remove the `public` modifier on methods in those classes

### CODE REVIEW

File: google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/SubscriberStats.java [hunk 1]
```diff
@@ -27,52 +27,52 @@
 
   @AutoValue
   // TODO: Finish implementation.
-  public abstract static class Stats {}
+  abstract static class Stats {}
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/SubscriberStats.java]: To confirm, everything in this class is only package private? Want to make sure we won't break any existing implementations.

Comment (sduskis) [reply google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/SubscriberStats.java]: correct.  The top level class `SubscriberStats` is package private, so no one can access the `Stats` inner class from outside the package.

**Comment by kolea2** [comment]
This LGTM, but I had a question regarding the MessageDispatcher change in https://github.com/googleapis/google-cloud-java/pull/4949. It looks like this change made it into both PRs, can you remove it from this once, I'll approve, and we can follow up on the other PR separately?

**Comment by sduskis** [comment]
@kolea2, done.

**Pull request merged**